brainstormforce/astra
# AST-3422 - Feature Secondary Button Outline Styling Options.

### Description

User Story:

- As a PO/End-user
- I want to style my site’s outline buttons
- Add design styling options which will be working for outline buttons
- Make this functionality extendable 
- We will cover wp outline buttons by core but this styling should also work on components/elements where “ast-outline-button“ class will be use.
- In this way we provide users extendability to have this option usage in their way & custom buttons.
- So that  it will be differentiating from solid buttons, and the side-by-side buttons looks elegant.

- JIRA - https://brainstormforce.atlassian.net/jira/software/projects/AST/boards/10?assignee=62c6a337ce5a604dbfb504b8&selectedIssue=AST-3422

### Screenshots

https://d.pr/i/fgH1ZJ

### Types of changes




New Feature

### How has this been tested?

- Two new tabs should be visible - Primary, Secondary in Global > Button.
- Create a new Button with its WordPress Styling set to “Outline”.
- For this button, all the new Secondary Options should work as expected.
- Note: If Secondary Options are set to empty or blank, then some of the values will be inherited from Primary section (because earlier as we had a common setting for all the buttons: Solids + Outline)
If we set any Secondary Options they should apply as expected in Customizer, Frontend and the Editor.
- Older Users Compatibility :- Switch to previous version of Astra (master), and set up some options for Buttons and now switch to the latest version, here you will observe couple of things:
             a. Buttons should remain as it in frontend (no button styling should break for Solids as well as Outline)
             b. Button presets will be reset in customizer will be reset, as now Button Presets are divided in two parts Primary and Secondary, so they will apply as per whatever value we select and will only work for that type, either Primary or Secondary.
- Create a new Custom HTML Block - Add a button and a class “ast-outline-button”. Secondary option values should also apply to this button.
- Test by installing some Starter templates on older version, switch to new version (latest) and ensure that no button styles are breaking.
- Test with Surecart as well.
- Test Responsive options.

### Checklist:
